### DISCUSSION TITLE
Incorrect Typescript example for RecipeVariantProps

### DISCUSSION THREAD

**Discussion by jorqensen** [chakra-ui/panda#2905] [General]
### Subject

TypeScript guide for recipe variants

### Description

Seems to be a mistake in the docs regarding [`RecipeVariantProps`](https://panda-css.com/docs/concepts/recipes#typescript-guide).

Using `RecipeVariantProps` results in this type:

```ts
type ButtonVariants = {
    visual?: "primary" | "secondary" | undefined;
} | undefined
```

Which is not the expected type, according to the documentation:

`// { size?: 'small' | 'large' }`

However, using `RecipeVariant` gives an output closer to what is "expected":

```ts
type ButtonVariants = {
    visual: "primary" | "secondary";
}
```

The type still doesn't mark the `visual` as optional, as the docs reference. Seems to be expected behavior according to #2182 however.

Could we update the docs to either mention `RecipeVariant` or both with examples? If a PR is needed, let me know.

**Comment by segunadebayo** [comment]
Good point. Feel free to open a PR for this @jorqensen 

I think we should mention both types.

**Reply by jorqensen** [reply]
Sorry, I've only just seen this! PR should be ready for review.
